src/rpc/blockchain.cpp

```diff
@@ -232,6 +232,13 @@ UniValue blockToJSON(const CBlock& block, const CBlockIndex* blockindex, bool tx
     result.pushKV("version", block.nVersion);
     result.pushKV("merkleroot", block.hashMerkleRoot.GetHex());
     result.pushKV("finalsaplingroot", blockindex->hashFinalSaplingRoot.GetHex());
+    {
+        SaplingMerkleTree saplingTree;
+        pcoinsTip->GetSaplingAnchorAt(blockindex->hashFinalSaplingRoot, saplingTree);
+        CDataStream s(SER_NETWORK, PROTOCOL_VERSION);
+        s << saplingTree;
+        result.pushKV("treestate", HexStr(s.begin(), s.end()));
```

Comment:
We need to use `GetSproutAnchorAt()`, maybe have 2 output fields `saplingtreestate` and `sproutstreestate` ?